### ISSUE TITLE
Training problem on higher version Pytorch

### ISSUE THREAD

**Issue by lcybuzz** [liutinglt/CE2P#13]
I tried to train CE2P on Pytorch 0.5. But this error occurs: 
> dimension specified as 0 but tensor has no dimensions

To solve this,  I modified `forward()` in `CriterionDataParallel` and added a dimension for each entry in `outputs`. Then the training works and the loss has successfully converged.

I opened this issue because the loss failed to converge on Pytorch 0.5. Now the training seems working fine. So I think this issue should be closed. Thanks a lot!
